sql/tablewriter.go

```diff
@@ -641,39 +639,39 @@ func (td *tableDeleter) deleteAllRowsScan() error {
 			// Done deleting rows.
 			break
 		}
-		_, err = td.row(row)
+		_, err = td.row(ctx, row)
 		if err != nil {
 			return err
 		}
 	}
-	return td.finalize()
+	return td.finalize(ctx)
 }
 
 // deleteIndex runs the kv operations necessary to delete all kv entries in the
 // given index. This may require a scan.
-func (td *tableDeleter) deleteIndex(idx *sqlbase.IndexDescriptor) error {
+func (td *tableDeleter) deleteIndex(ctx context.Context, idx *sqlbase.IndexDescriptor) error {
 	if len(idx.Interleave.Ancestors) > 0 || len(idx.InterleavedBy) > 0 {
 		if log.V(2) {
-			log.Info(context.TODO(), "delete forced to scan: table is interleaved")
+			log.Info(ctx, "delete forced to scan: table is interleaved")
 		}
-		return td.deleteIndexScan(idx)
+		return td.deleteIndexScan(ctx, idx)
 	}
-	return td.deleteIndexFast(idx)
+	return td.deleteIndexFast(ctx, idx)
 }
 
-func (td *tableDeleter) deleteIndexFast(idx *sqlbase.IndexDescriptor) error {
+func (td *tableDeleter) deleteIndexFast(ctx context.Context, idx *sqlbase.IndexDescriptor) error {
 	indexPrefix := sqlbase.MakeIndexKeyPrefix(td.rd.helper.tableDesc, idx.ID)
 	indexStartKey := roachpb.Key(indexPrefix)
 	indexEndKey := indexStartKey.PrefixEnd()
 
 	if log.V(2) {
-		log.Infof(context.TODO(), "DelRange %s - %s", indexStartKey, indexEndKey)
+		log.Infof(ctx, "DelRange %s - %s", indexStartKey, indexEndKey)
```

Comment:
Can you file an issue?

On Jul 28, 2016 11:02, "Daniel Harrison" <notifications@github.com> wrote:

> In sql/tablewriter.go
> <https://github.com/cockroachdb/cockroach/pull/8074#discussion_r72639455>:
>
> >  	indexPrefix := sqlbase.MakeIndexKeyPrefix(td.rd.helper.tableDesc, idx.ID)
> >  	indexStartKey := roachpb.Key(indexPrefix)
> >  	indexEndKey := indexStartKey.PrefixEnd()
> >
> >  	if log.V(2) {
> > -		log.Infof(context.TODO(), "DelRange %s - %s", indexStartKey, indexEndKey)
> > +		log.Infof(ctx, "DelRange %s - %s", indexStartKey, indexEndKey)
>
> Yeah, it was a TODO (
> https://github.com/cockroachdb/cockroach/pull/8074/files/86e145a75b329e378d7c68fec15081401c9ff87d#diff-ba1f4a7dd17d7c48d43a1e60bd2cd1b0R192).
> This commit was a mindless refactor, where that still has a couple things I
> haven't sat down and figured out (especially making it play well with
> vmodule, I use that constantly while doing sql work), so I'd like to keep
> that separate if it's okay
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/8074/files/86e145a75b329e378d7c68fec15081401c9ff87d#r72639455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABdsPHQKbSI7hAc9Er_PorCUkYvSwEAFks5qaMRtgaJpZM4JWpu9>
> .
>